app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/MessagesMasterFragment.java

```diff
@@ -0,0 +1,108 @@
+ * interface.
+ */
+public class MessagesMasterFragment extends BaseFragment<GGApplication>
+        implements MessagesRecyclerViewAdapter.OnItemClickListener, MessageListViewModel.OnDataChangedListener {
+
+    private OnMessageMasterFragmentClickListener mListener;
+    private RecyclerView mRecyclerView;
+    private MessagesRecyclerViewAdapter mAdapter;
+    private MessageListViewModel mMessagesViewModel;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_messages_master_list, container, false);
+
+
+        // Set the adapter
+        if (view instanceof RecyclerView) {
+            Context context = view.getContext();
+            mRecyclerView = (RecyclerView) view;
+            mRecyclerView.setLayoutManager(new LinearLayoutManager(context));
+            mMessagesViewModel = new MessageListViewModel(this);
+            mAdapter = new MessagesRecyclerViewAdapter(mMessagesViewModel.getRandomAccessor(), this);
+            mRecyclerView.setAdapter(mAdapter);
+        }
+        return view;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+//        if (context instanceof OnMessageMasterFragmentClickListener) {
+//            mListener = (OnMessageMasterFragmentClickListener) context;
+//        } else {
+//            throw new RuntimeException(context.toString()
+//                    + " must implement OnItemClickListener");
+//        }
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
+        mListener = null;
+    }
+
+    @Override
+    protected int getFragmentLayout() {
+        return R.layout.fragment_messages_master_list;
+    }
+
+    @Override
+    public void onListItemInteraction(Message item) {
+        Toast.makeText(getActivity(), item.getMessageId(), Toast.LENGTH_SHORT).show();
+        mMessagesViewModel.addMessage(DummyMessagesContent.createDummyItem(new Random().nextInt() % 100));
+//        mListener.onListFragmentInteraction(item);
+    }
+
+    @Override
+    public void onDataChanged() {
+        mAdapter.notifyDataSetChanged();
+    }
+
+    /**
+     * This interface must be implemented by activities that contain this
+     * fragment to allow an interaction in this fragment to be communicated
+     * to the activity and potentially other fragments contained in that
+     * activity.
+     * <p/>
+     * See the Android Training lesson <a href=
+     * "http://developer.android.com/training/basics/fragments/communicating.html"
+     * >Communicating with Other Fragments</a> for more information.
+     */
+    public interface OnMessageMasterFragmentClickListener {
```

Comment:
If we're to do MVVM, there wont be any direct interaction between the Fragment and its container. Right ?